rules/order.js

```diff
@@ -1,6 +1,181 @@
 'use strict';
 
 // See https://github.com/hudochenkov/stylelint-order
+
+const order = [
+    // Overview
+    'all',
+    'content',
+    'box-sizing',
+    'will-change',
+
+    // Position, dimensions and boundaries
+    'position',
+    'top',
+    'right',
+    'bottom',
+    'left',
+
+    'width',
+    'min-width',
+    'max-width',
+    'height',
+    'min-height',
+    'max-height',
+
+    'z-index',
+    'overflow',
+    'overflow-x',
+    'overflow-y',
+
+    ...trbl('margin'),
+    ...trbl('padding'),
+
+    'transform',
+    'zoom',
+
+    // Display
+    'display',
+
+    // Flexbox only
+    'flex',
+    'flex-basis',
+    'flex-grow',
+    'flex-shrink',
+    'flex-flow',
+    'flex-wrap',
+    'flex-direction',
+    'justify-content',
+    'align-self',
+    'align-items',
+    'align-content',
+    'order',
+
+    // Grid Layout
+    'grid',
+    'grid-area',
+    'grid-auto-columns',
+    'grid-auto-flow',
+    'grid-auto-rows',
+    'grid-gap',
+    'grid-column',
+    'grid-column-end',
+    'grid-column-gap',
+    'grid-column-start',
+    'grid-row',
+    'grid-row-end',
+    'grid-row-gap',
+    'grid-row-start',
+    'grid-template',
+    'grid-template-areas',
+    'grid-template-columns',
+    'grid-template-rows',
+
+    'clear',
+    'float',
+    'visibility',
+    'vertical-align',
+    'direction',
+
+    // Table related stuff only
+    'table-layout',
+    'empty-cells',
+
+    // Asthethic
+    'list-style',
+    ...trbl('border'),
```

Comment:
What if we use https://github.com/gilmoreorless/css-shorthand-properties?

It seems that that's exactly what we need, plus it works for a lot of other properties that are missing or manually declared. If you agree, could you use this module function for every property that it's supported?

Note that the properties order that this module returns is different for some properties, which is a breaking change. I'm fine with it, but be sure to include it in the commit body. You will probably also need to update a few test fixtures

Reply:
There's also https://github.com/mahirshah/css-property-parser which has `getShorthandComputedProperties`. It seems a more robust package because it uses that from [MDN](https://github.com/mdn/data),